server/bin/pbench-backup-tarballs

```diff
@@ -1,83 +1,123 @@
-        echo "$PROG: rsync failed with code: $rsync_sts, $rlb"
-    fi
-fi
-
-echo "end-$(timestamp)"
-
-log_finish
-
-# Back up some way and find the empty line separating the list of files
-# from the stats.
-cat << EOF > $index_content
-$PROG.$TS($PBENCH_ENV)
-EOF
-
-if [ -e $stats ] ;then
-    tail -n 20 $stats | sed -n '/^$/,$p' >> $index_content
-fi
-
-pbench-report-status --name $PROG --timestamp $TS --type status $index_content
-
-exit 0
+    archive = "{}/".format(config.ARCHIVE)
+    stat_out = sh.rsync("-va", "--stats", exclude, archive, config.BACKUP)
+    if stat_out.exit_code == 0:
+        _logger.info(str(stat_out))
+    elif stat_out.exit_code != 0:
+        _logger.warning("Failed to do rsync between {} and {}: {}\n".format(config.ARCHIVE, config.BACKUP, e))
+        report.write("Failed to do rsync between {} and {}: {}\n".format(
+            config.ARCHIVE, config.BACKUP, e))
+
+def backup_data_to_s3(s3_obj, config, report):
+    # Function to copy data to s3
+    try:
+        tarlist = glob.glob('{}/*/*.tar.xz'.format(config.ARCHIVE))
+        s3_obj.upload_to_bucket(tarlist, report, _logger)
+    except Exception as e:
+        _logger.warning("Failed to upload data from {} to s3: {}\n".format(config.ARCHIVE, e))
+        report.write("Failed to upload data from {} to s3: {}\n".format(config.ARCHIVE, e))
+
+def main(options):
+    if not options.cfg_name:
+        print("{}: ERROR: No config file specified; set CONFIG env variable or"
+                " use --config <file> on the command line".format(_NAME_),
+                file=sys.stderr)
+        return 2
+
+    try:
+        config = PbenchConfig(options.cfg_name)
+    except BadConfig as e:
+        print("{}: {}".format(_NAME_, e), file=sys.stderr)
+        return 3
+
+    # add a BACKUP field to the config object
+    config.BACKUP =config.conf.get("pbench-server", "pbench-backup-dir")
+
+    global _logger
+    _logger = get_pbench_logger(_NAME_, config)
+
+    # call the S3Config class
+    s3_obj = S3Config(config)
+
+    _logger.info('start-{}'.format(s3_obj.connector.ts()))
+
+    prog = os.path.basename(sys.argv[0])
+    report_dir = "{}/{}.{}".format(config.TMP, prog, os.getpid())
+
+    try:
+        os.mkdir(report_dir)
+    except FileExistsError:
+        # directory already exists, ignore
+        pass
+    report = "{}/report".format(report_dir)
+
+    with open(report, 'a') as reportf:
+        # call the rsync function
+        rsync_archive_backup(config, reportf)
+
+        # call the backup to s3 function
+        backup_data_to_s3(s3_obj, config, reportf)
+
+    with open(report, 'r+') as f:
+        TS = s3_obj.connector.ts()
+        content = f.read()
+        f.seek(0, 0)
+        f.write("{}.{}({})\n".format(prog, TS, config.PBENCH_ENV))
+        f.write(content)
+
+    _logger.info('end-{}'.format(s3_obj.connector.ts()))
+
+    # Call pbench-report-status
+    subprocess.call(['./pbench-report-status', '--name={}'.format(prog),
```

Comment:
This is not going to work unless you run the script from the bin directory. I think you should leave the `./` out and just make sure that the PATH includes the appropriate directory.